modules/lib/src/main/resources/assets/js/app/browse/filter/AggregationsProcessor.ts

```diff
@@ -0,0 +1,141 @@
+import {Aggregation} from 'lib-admin-ui/aggregation/Aggregation';
+import {Bucket} from 'lib-admin-ui/aggregation/Bucket';
+import {BucketAggregation} from 'lib-admin-ui/aggregation/BucketAggregation';
+import {WorkflowState} from 'lib-admin-ui/content/WorkflowState';
+import {i18n} from 'lib-admin-ui/util/Messages';
+import {ContentAggregations} from './ContentAggregations';
+import {GetPrincipalsByKeysRequest} from '../../security/GetPrincipalsByKeysRequest';
+import * as Q from 'q';
+import {PrincipalKey} from 'lib-admin-ui/security/PrincipalKey';
+import {Principal} from 'lib-admin-ui/security/Principal';
+import {StringHelper} from 'lib-admin-ui/util/StringHelper';
+import {GetLocalesRequest} from '../../resource/GetLocalesRequest';
+import {Locale} from 'lib-admin-ui/locale/Locale';
+
+export class AggregationsProcessor {
+
+    private principals: Map<string, string> = new Map<string, string>();
+    private locales: Locale[];
+    private readonly currentUserId: string;
+
+    constructor(userId: string) {
+        this.currentUserId = userId;
+    }
+
+    updateWorkflowAggregations(aggregations: Aggregation[], total: number) {
+        const workflowAggr: Aggregation = aggregations.find((aggr: Aggregation) => aggr.getName() === ContentAggregations.WORKFLOW);
+
+        if (workflowAggr) {
+            this.updateWorkflowAggregation(<BucketAggregation>workflowAggr, total);
+        }
+    }
+
+    private updateWorkflowAggregation(workflowAggr: BucketAggregation, total: number): void {
+        // contents might not have a workflow property, thus aggregation won't see those contents, but they are treated as ready
+        const inProgressKey: string = WorkflowState[WorkflowState.IN_PROGRESS].toLowerCase();
```

Comment:
`WorkflowState[WorkflowState.IN_PROGRESS].toLowerCase()`

As we seem to use this all over the place and since `Workflow.getStateAsString` is not used at the moment, I suggest we change it like this:

```Typescript
    static getStateAsString(state: WorkflowState): string {
        return WorkflowState[state].toLowerCase();
    }
```

and start using `Workflow.getStateAsString(WorkflowState.IN_PROGRESS)`

Reply:
Yeah, this is even better. I assume we are not sending numeric enum values to the API.